### DISCUSSION TITLE
Help Understanding Parallelism and Cronjob

### DISCUSSION THREAD

**Discussion by ericwyles** [renovatebot/renovate#31562] [Request Help] [answered]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

gitlab, renovate 38.67.2

### Please tell us more about your question or problem

Hi, I am working on deploying renovate self hosted on kubernetes using the [helm chart](https://github.com/renovatebot/helm-charts). My target environment is air-gapped with self hosted gitlab also running in Kubernetes.

I'm trying to understand how the parallelism works if I run renovate cronjob with parallel pods: https://github.com/renovatebot/helm-charts/blob/5feff0d003526a90ab7688b71e3cd19ad5c99d2c/charts/renovate/values.yaml#L47

If using this configuration with > 1 pods, do the pods coordinate through cache or some other mechanism to subdivide the work? I'm looking for options that will allow repos to be processed in parallel, but I'm having trouble understanding how the renovate bot itself functions if multiple pods are running at the same time. Is there a section of doc I can read that explains this? I have my test renovate connected to an in-cluster instance of Valkey... will the pods coordinate via that cache to avoid duplicating work?

Thanks!

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment, marked as answer]
The Renovate CLI you are running has no inter-process coordination,so you can't just scale up pods and hope magic will happen. You would need to build your own orchestration/scheduling concept to try to avoid processes competing with each other on the same repo.

For an officially supported variant of Renovate with parallelism, please see the "Mend Renovate Enterprise" product via mend.io/renovate

**Reply by ericwyles** [reply]
Thank you @rarkins this is exactly the info I needed.
